File: src/main/java/com/jhl/mds/dto/RedisServerDTO.java
package com.jhl.mds.dto;

import com.jhl.mds.dao.entities.RedisServer;
import com.jhl.mds.jsclientgenerator.JsClientDTO;
import lombok.*;
import org.springframework.stereotype.Component;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(exclude = {"serverId", "name"})
public class RedisServerDTO {
    private int serverId;
    private String name;
    private String host;
    private String port;
    private String username;
    private String password;

    @Component
    public static class Converter {

        public RedisServerDTO from(RedisServer server) {
            return RedisServerDTO.builder()
                    .serverId(server.getServerId())
                    .name(server.getName())
                    .host(server.getHost())
                    .port(server.getPort())
                    .username(server.getUsername())
                    .password(server.getPassword())
                    .build();
        }

        public RedisServer toDAO(RedisServerDTO dto) {
            return RedisServer.builder()
                    .serverId(dto.getServerId())
                    .name(dto.getName())
                    .host(dto.getHost())
                    .port(dto.getPort())
                    .username(dto.getUsername())
                    .password(dto.getPassword())
                    .build();
        }
    }
}

File: src/main/resources/static/resources/js/source/dto/redis-server-dto.js
// @flow
export class RedisServerDTO {

    /**
     * @type {number}
     */
    serverId : ?number = 0;
    /**
     * @type {string}
     */
    name : ?string = '';
    /**
     * @type {string}
     */
    host : ?string = '';
    /**
     * @type {string}
     */
    port : ?string = '';
    /**
     * @type {string}
     */
    username : ?string = '';
    /**
     * @type {string}
     */
    password : ?string = '';

    /**
     * @param serverId {number}
     * @param name {string}
     * @param host {string}
     * @param port {string}
     * @param username {string}
     * @param password {string}
     */
    constructor(serverId : ?number, name : ?string, host : ?string, port : ?string, username : ?string, password : ?string) {
        this.serverId = serverId;
        this.name = name;
        this.host = host;
        this.port = port;
        this.username = username;
        this.password = password;
    }


}
